src/main/java/pl/marekjava/App.java

```diff
@@ -49,12 +49,31 @@ private static void playGame(List<Player> players) {
                         guessLetter(password, input, player);
                     } else {
                         isRoundContinue = !guessPassword(input, player);
-                        if (!isRoundContinue) break;
                     }
+                    sumOfPoints.put(player, player.getPoints());
+                    if (!isRoundContinue) break;
                     isRoundContinue = !checkPassword();
+                    if (!isRoundContinue) break;
                 }
             }
         }
+
+        Map<Player, Integer> newMapSortedByValue = getReversedSortedMap(sumOfPoints);
```

Comment:
Rozwiązanie z mapą ok, ale wydaje mi się nieco przekombinowane. Skoro każdy z graczy ma pole z punktami, nie łatwiej posortować już istniejącą listę graczy za pomocą odpowiedniego komparatora? Możesz w ramach praktyki sprawdzić.
[](https://www.baeldung.com/java-8-sort-lambda), a konkretnie kawałek 'Now let's see how we can use a custom Comparator with the sorted() API'  i 'We can simplify the above example even further if we use the Comparator.comparing() method'

Reply:
Ok. Poprawiłem